src/main/java/com/minecolonies/colony/buildings/BuildingMiner.java

```diff
@@ -80,10 +142,21 @@
      */
     private boolean foundLadder = false;
 
-    //Stores the levels of the miners mine. This could be a map<depth,level>
+    /**
+     * Stores the levels of the miners mine. This could be a map<depth,level>
+     */
     private List<Level> levels = new ArrayList<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private field 'levels' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)